src/nvim/terminal.c

```diff
@@ -229,7 +237,8 @@ static void schedule_termrequest(Terminal *term, char *payload, size_t payload_l
   term->pending.send = xmalloc(sizeof(StringBuilder));
   kv_init(*term->pending.send);
   multiqueue_put(main_loop.events, emit_termrequest, term, payload, (void *)payload_length,
-                 term->pending.send);
+                 term->pending.send, (void *)(intptr_t)term->cursor.row,
+                 (void *)(intptr_t)term->cursor.col);
```

Comment:
This doesn't completely work. In some cases, vterm seems to call the `on_osc` callback before it's updated its cursor position, so it reports the wrong position. This seems to happen more frequently when there is a lot of text (e.g. `ls -al` vs just `ls`).